packages/frontend/src/redux/slices/account/createAccountThunks.js

```diff
@@ -219,3 +228,70 @@ export const finishLocalSetupForZeroBalanceAccount = createAsyncThunk(
 
     }
 );
+
+export const initiateSetupForZeroBalanceAccountPhrase = createAsyncThunk(
+    `${SLICE_NAME}/initiateSetupForZeroBalanceAccountPhrase`,
+    async ({
+        implicitAccountId,
+        recoveryKeyPair
+    }, { dispatch }) => {
+        try {
+            try {
+                await sendJson('POST', `${ACCOUNT_HELPER_URL}/account/seedPhraseAdded`, {
+                    accountId: implicitAccountId,
+                    publicKey: recoveryKeyPair.publicKey.toString()
+                });
+            } catch (e) {
+                if (e.status === 400) {
+                    console.log(`Public key ${recoveryKeyPair.publicKey.toString()} has previously been added as recovery method to account. Continueing setup...`);
+                } else {
+                    throw new WalletError(e, 'initiateSetupForZeroBalanceAccountPhrase.error');
+                }
+            }
+            await wallet.importZeroBalanceAccount(implicitAccountId, recoveryKeyPair);
+        } catch (e) {
+            dispatch(showCustomAlert({
+                success: false,
+                messageCodeHeader: 'error',
+                messageCode: 'walletErrorCodes.initiateZeroBalanceAccount',
+                errorMessage: e.message
+            }));
+        }
+
+    }
+);
+
+export const initiateSetupForZeroBalanceAccountLedger = createAsyncThunk(
+    `${SLICE_NAME}/initiateSetupForZeroBalanceAccountLedger`,
+    async ({
+        implicitAccountId,
+        ledgerPublicKey,
+        ledgerHdPath
+    }, { dispatch }) => {
+        try {
+            try {
+                await sendJson('POST', `${ACCOUNT_HELPER_URL}/account/ledgerKeyAdded`, {
+                    accountId: implicitAccountId,
+                    publicKey: ledgerPublicKey.toString()
+                });
+            } catch (e) {
+                if (e.status === 400) {
+                    console.log(`Ledger public key ${ledgerPublicKey.toString()} has previously been added as recovery method to account. Continueing setup...`);
+                } else {
+                    throw new WalletError(e, 'initiateSetupForZeroBalanceAccountLedger.error');
+                }
+            }
+            await setKeyMeta(ledgerPublicKey, { type: 'ledger' });
+            await setLedgerHdPath({ accountId: implicitAccountId, path: ledgerHdPath });
```

Comment:
Previously we would check if the `path` exists before we call `setLedgerHdPath()`, is it safe to assume it will always have a value now?

Reply:
Also this function isn't `async` 😄

Reply:
Yep it has the default `ledgerHdPath` set in `SetupLedgerNewAccount` here: https://github.com/near/near-wallet/blob/master/packages/frontend/src/components/accounts/ledger/SetupLedgerNewAccount.js#L17 so it'll always have a path :)